packages/api2/src/periodsettings/periodsettings.service.ts

```diff
@@ -0,0 +1,110 @@
+import { ServiceException } from '@/shared/service-exception';
+import { UtilsProvider } from '@/utils/utils.provider';
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Request } from 'express';
+import { Model, Types } from 'mongoose';
+import { PeriodDocument } from 'src/periods/schemas/periods.schema';
+import { PeriodStatusType } from '../../../api/src/period/types';
+import {
+  PeriodSettings,
+  PeriodSettingsDocument,
+} from './schemas/periodsettings.schema';
+import { SetPeriodSettingDto } from './dto/set-periodsetting.dto';
+import { UploadedFile } from 'express-fileupload';
+
+@Injectable()
+export class PeriodSettingsService {
+  constructor(
+    @InjectModel(PeriodSettings.name)
+    private periodSettingsModel: Model<PeriodSettingsDocument>,
+    private periodModel: Model<PeriodDocument>,
+    private utils: UtilsProvider,
+  ) {}
+
+  async findAll(periodId: Types.ObjectId): Promise<PeriodSettings[]> {
+    const period = await this.periodModel.findById(periodId);
```

Comment:
I would like us to avoid using the model directly and instead use `periodService.findOneById()`.